internal/services/nginx/nginx_configuration_data_source.go

```diff
@@ -87,6 +82,29 @@ func (m ConfigurationDataSource) Attributes() map[string]*pluginsdk.Schema {
 			Computed: true,
 		},
 	}
+
+	if !features.FivePointOhBeta() {
+		dataSource["protected_file"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeSet,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Deprecated: "The `content` field is deprecated and will be removed in v5.0 of the AzureRM Provider.",
+						Type:       pluginsdk.TypeString,
+						Computed:   true,
+						Sensitive:  true,
+					},
+
+					"virtual_path": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		}
```

Comment:
It's helpful to target the `content` field specifically, makes it easier for the reviewer to see what will be removed in 5.0
```suggestion
		dataSource["protected_file"].Elem.(*pluginsdk.Resource).Schema["content"] = &pluginsdk.Schema{
			Type:     pluginsdk.TypeString,
			Computed: true,
			Sensitive: true,
			Deprecated: "the `content` property is deprecated and will be removed in v5.0 of the AzureRM Provider.",
		}
```

Reply:
thanks for sharing this 👍